core/src/main/java/com/google/cloud/sql/core/SqlAdminApiService.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.core;
+
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.services.sqladmin.SQLAdmin;
+import com.google.api.services.sqladmin.model.ConnectSettings;
+import com.google.api.services.sqladmin.model.GenerateEphemeralCertRequest;
+import com.google.api.services.sqladmin.model.GenerateEphemeralCertResponse;
+import com.google.api.services.sqladmin.model.IpMapping;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.OAuth2Credentials;
+import com.google.cloud.sql.ApiClientFactory;
+import com.google.common.base.CharMatcher;
+import com.google.common.io.BaseEncoding;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyPair;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class SqlAdminApiService {
```

Comment:
I'd be curious if we can create a single method that returns everything we need. Also, there's no need to expose an asynchronous API. We can expose a synchronous API and internally use threading to run the two HTTP requests concurrently.

Reply:
Landing this class will also need some good integration tests that hit the real API, too.

For testing edge cases, I'm curious if the generated SQLAdmin type has any helper methods for us.